### PR TITLE
Setup it tests for MarkLogic (though failing)

### PR DESCRIPTION
**Pull request by jedesah**

### CODE REVIEW

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala [hunk 1]
```diff
@@ -16,24 +16,59 @@
...
+  type Eff0[A] = Coproduct[
+                    KeyValueStore[ReadFile.ReadHandle, Process[Task, Vector[Data]], ?],
+                    KeyValueStore[WriteFile.WriteHandle, Unit, ?],
+                    A]
+  type Eff1[A] = Coproduct[MonotonicSeq, Eff0, A]
+  type Eff2[A] = Coproduct[Read[Client, ?], Eff1, A]
+  type Eff[A]  = Coproduct[Task, Eff2, A]
+
+  def inter(uri: ConnectionUri): Task[Eff ~> Task] = {
+    def dbClient: DatabaseClient = DatabaseClientFactory.newClient("localhost", 8000)
```
Comment (wemrysi) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Seems like we should be using the passed in `ConnectionUri` here, no? Also, for now I guess we can assume that XCC is being provided via the rest api so we can use the same port, but if we need both I'm not sure that will always be the case, may need to allow configuration of that via query params (i.e. `xccPort=1234&restPort=6789` or something) unless we're able to abandon rest completely and perform all operations via XCC.

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala [hunk 2]
```diff
@@ -16,24 +16,59 @@
...
-    : FileSystemDef[Free[S, ?]] = FileSystemDef.fromPF {
+  type Eff0[A] = Coproduct[
+                    KeyValueStore[ReadFile.ReadHandle, Process[Task, Vector[Data]], ?],
+                    KeyValueStore[WriteFile.WriteHandle, Unit, ?],
+                    A]
+  type Eff1[A] = Coproduct[MonotonicSeq, Eff0, A]
+  type Eff2[A] = Coproduct[Read[Client, ?], Eff1, A]
+  type Eff[A]  = Coproduct[Task, Eff2, A]
+
+  def inter(uri: ConnectionUri): Task[Eff ~> Task] = {
```
Comment (wemrysi) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Does the `DatabaseClient` need to be shutdown when marklogic is unmounted? If so we should return a shutdown action along with the interpreter so mounting can cleanup appropriately.

**Comment by wemrysi** [comment]
It is ok if it doesn't, but the description says this sets up the `it` tests, but I don't see that anywhere (or is that already implemented in a previous commit to the branch?)

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala (hunk 1 above)
Comment (jedesah) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Ya, I was having trouble with the passed in `uri` so I did this to debug. I can spend time getting this correct if you want before merging, I don't think it should be too hard. (assuming `java.net.URI` has `getHost` and `getPort` or something)

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala (hunk 2 above)
Comment (jedesah) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Looks like it `client.release()`. Is that what the `Free.point()` argument down there is supposed to be doing? I copied that part from Postgres without understanding all arguments.

Comment (jedesah) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Then again, unclear if we will need `DatabaseClient` at all. Which is why I am moving ahead with `it` testing before polishing this code.

Comment (wemrysi) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Yeah, `DefinitionResult` contains an interpreter and a "close" action to be called when the filesystem is unmounted.

As for trying to make all of this really polished, yeah, I'm not saying it has to be perfect, though in this case adding the shutdown action isn't too big of a deal.

**Comment by jedesah** [comment]
Ya, parts of it were done in a previous commit. It just immediately hit a `???`. Now it's actually exercising the faulty interpreters.

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala (hunk 2 above)
Comment (jedesah) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: Agreed.

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala (hunk 1 above)
Comment (wemrysi) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala]: I'd say either correct it, or just hardcode the URI and don't even accept an argument, just to make it clear that you can't pass anything useful to the function, if/when we support the URI as an argument, we can add it.

**Comment by jedesah** [comment]
@wemrysi Pushed a commit that does some minor polishing. Let me know if good to merge in your opinion.

**Comment by wemrysi** [comment]
:shipit:

**Pull request merged**